src/utils/transition-manager.js

```diff
@@ -132,15 +134,19 @@ export default class TransitionManager {
       isRotating: startProps.bearing !== endProps.bearing ||
         startProps.pitch !== endProps.pitch
     };
-
+    const currentTime = Date.now();
+    const completion = this.state.interruption === TRANSITION_EVENTS.UPDATE ?
+                      (currentTime - this.state.startTime) / startProps.transitionDuration : 0;
+    const totalCompletion = this.state.totalCompletion + completion;
     this.state = {
       // Save current transition props
-      duration: endProps.transitionDuration,
+      totalCompletion,
+      duration: endProps.transitionDuration * (1 - totalCompletion),
```

Comment:
UPDATE mode needs to use duration, easing and interpolators from the previous transition.